#! /bin/sh /usr/share/dpatch/dpatch-run
## 19_gpsfake_fuser.dpatch by Tilman Koschnick <til@subnetz.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Work around a bug in fuser (see <http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=361639>).

@DPATCH@
diff -urNad gpsd~/gpsfake.py gpsd/gpsfake.py
--- gpsd~/gpsfake.py	2006-06-09 17:59:41.000000000 +0200
+++ gpsd/gpsfake.py	2006-06-10 15:07:43.250323520 +0200
@@ -200,7 +200,7 @@
         "Is the slave device of this pty opened?"
         if self.verbose:
             sys.stderr.write("slave_is_open() begins")
-        isopen = os.system("fuser -s " + self.slave) == 0
+        isopen = os.system("fuser " + self.slave + " > /dev/null 2>&1") == 0
         if self.verbose:
             sys.stderr.write("slave_is_open() ends")
         return isopen
